### PR TITLE
Bug 1636498 - Refactor goto / sticky lines interaction.

### PR DESCRIPTION
**Pull request by emilio**

### CODE REVIEW

**Comment by staktrace** [comment]
This looks ok to me but @asutherland should probably take a look as he understands the intricacies of this much better than I do.

File: static/css/mozsearch.css [hunk 1]
```diff
@@ -262,64 +259,28 @@ body {
      .source-line-with-number div necessary for sticky operation. */
   position: relative;
```
Comment (asutherland) [inline static/css/mozsearch.css:260]: It seems like maybe we can remove this `position: relative` and the comment that suggest it's necessary to create a containing block.  Per bullet 1 of https://developer.mozilla.org/en-US/docs/Web/CSS/Containing_Block#Identifying_the_containing_block it seems like the fact that this is `display: block` is sufficient?  (Turning off the `position: relative` certainly doesn't break anything on the current searchfox.org.)

Comment (emilio) [reply static/css/mozsearch.css]: Yes, you're right! In fact `display: block` is not even needed, as it's the default for `<div>`s.

File: static/css/mozsearch.css [hunk 2]
```diff
@@ -606,21 +567,15 @@ span[data-symbols].hovered {
...
-
-       The negative top could be eliminated and replaced with use of the CSS
-       scroll-padding mechanism introduced in Firefox 68 (currently beta).
-       However, we can't actually get rid of the synthetic anchor.  And since
-       scroll-padding isn't on release yet, we're sticking with this for now. */
-    top: calc(var(--max-sticky-lines) * 12px * -1.3);
-    /* and for the negative top to work, we need to be position: absolute and
-       .line-number needs to be position: relative to create a containing block
-       that we can offset against. */
-    position: absolute;
```
[the remark this answers is not in the document]
Comment (emilio) [reply static/css/mozsearch.css:623]: Indeed, done :)

File: static/js/code-highlighter.js [hunk 3]
```diff
@@ -230,7 +228,7 @@ var Highlight = new (class Highlight {
       }
     }
 
-    let line = parseInt(event.target.id.substring(1), 10);
+    let line = parseInt(event.target.innerText, 10);
```
[the remark this answers is not in the document]
Comment (emilio) [reply static/js/code-highlighter.js]: Indeed! :(

However! I found a way to get rid of the nasty hack, and just be consistent across browsers.

See https://github.com/mozsearch/mozsearch/pull/327/commits/570ef6f1ca277f06c4168d6cf644267c8834f31c. The TLDR is that I recently fixed find-in-page to be able to search for pseudo-elements in https://bugzilla.mozilla.org/show_bug.cgi?id=1627643, so there's no need to make them user-select: none explicitly anymore, if we take a little regression for non-Nightly / non-Beta Firefox.

**Comment by emilio** [comment]
I pushed a few more commits based on your feedback, can you take a look @asutherland?

**Pull request merged**